server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts

```diff
@@ -582,6 +585,19 @@ export class SummaryWriter implements ISummaryWriter {
         pending: ISequencedOperationMessage[]): Promise<ISequencedDocumentMessage[]> {
         if (lt - gt <= 1) {
             return [];
+        } else if (pending.length > 0) {
+            // Check for logtail ops in the unprocessed ops currently present in memory
+            const pendingOps = pending.filter((op) => op.operation.sequenceNumber > gt && op.operation.sequenceNumber < lt).map((op) => op.operation);
+            if (pendingOps.length > 0 && pendingOps[0].sequenceNumber === gt + 1 && pendingOps[pendingOps.length - 1].sequenceNumber === lt - 1) {
```

Comment:
Is it possible that there be a gap in pendingOps, so even if the condition above is true, we might still have to fetch from the DB/get Deltas via Alfred?

Reply:
I think that is very unlikely. Pending ops is the list of ops between two summaries that we store in memory. Since we are reading the ordered ops from kafka and populating pendingops, the ordering is guaranteed. What could happen is that the pending ops could be lost on scribe restart but it will read from the correct offset with the correct ordering post restart.